src/geofabrics/dem.py

```diff
@@ -1670,62 +1754,158 @@ def _create_data_set(
             # ensure the expected CF conventions are followed
             self._write_netcdf_conventions_in_place(dem, self.catchment_geometry.crs)
             dems.append(dem)
+
         if len(dems) == 1:
             dem = dems[0]
         else:
             dem = rioxarray.merge.merge_datasets(dems, method="first")
+
+        # After merging LiDAR datasets set remaining NaN to no data/LiDAR
         # data_source: set areas with no values to No Data
-        dem.data_source.data[
-            numpy.isnan(dem.data_source.data)
-        ] = self.SOURCE_CLASSIFICATION["no data"]
+        dem.data_source.data = xarray.where(
+            dem.data_source.notnull(),
+            dem.data_source,
+            self.SOURCE_CLASSIFICATION["no data"],
+            keep_attrs=True,
+        )
```

Comment:
hum... shouldn't it be `dem["data_source"] = xarray.where(...` ?

Reply:
Yea _ I reckon you're right... good thing it seems to have been cleaver enough to work it out. Will fix for sure though